src/components/profiles-modals/RenameProfileModal.vue

```diff
@@ -0,0 +1,82 @@
+<script lang="ts">
+import { Vue, Component } from 'vue-property-decorator';
+import { ModalCard } from "../all";
+import sanitize from 'sanitize-filename';
+import R2Error from "../../model/errors/R2Error";
+
+@Component({
+    components: {ModalCard}
+})
+export default class RenameProfileModal extends Vue {
+    private newProfileName: string = '';
+
+    get isOpen(): boolean {
+        this.newProfileName = this.$store.getters['profile/activeProfileName'];
```

Comment:
I'm a bit anxious reading this value from Vuex in here.

- isOpen can be called multiple times as things get rerendered. Especially if the modal is closed, this is wasted effort
- at least in theory this can lead to situations where `this.newProfileName` is overwritten with a value from Vuex e.g. while user is editing the name in the input

However I'm not entirely sure how to fix this:

- We can't update `newProfileName` in Vue lifecycle hook like `created()` since that won't update the value when user selects a different profile
- We can't use `get`/`set` computed values to access active profile name since we don't want to update the value as the user types in the input
- Some modals (e.g. DownloadModModal) stores extra parameters that are updated when the modal is opened in the ModalsProfile, but that's for modals that don't have a better place for storing the values, while here we have the ProfilesModule

Best I can come up with is this:

```typescript
    @Watch('$store.state.profile.activeProfile')
    activeProfileChanged(newProfile: Profile, oldProfile: Profile|null) {
        if (oldProfile === null || oldProfile.getProfileName() === this.newProfileName) {
            this.newProfileName = newProfile.getProfileName();
        }
    }
```

- We're watching for changes to `activeProfile` (we can't watch `activeProfileName` since it's a getter and not in the state)
- If `oldProfile` was null (profile was first selected) we update `this.newProfileName`
- Or if `oldProfile` matched `this.newProfileName` we update the value (we don't want to overwrite changes the changes the user has made)
- The part I don't like about this solution is that it somewhat exposes too much of the Vuex side to the Vue component

Thoughts?

Reply:
I think it's ok, can't come up with anything better and I think this is exaclty the kind of thing that `Watch` exists for.